src/abi/linux_abi.rs

```diff
@@ -576,21 +592,24 @@ impl Attr {
             atimensec: st.st_atime_nsec as u32,
             mtimensec: st.st_mtime_nsec as u32,
             ctimensec: st.st_ctime_nsec as u32,
-            mode: st.st_mode,
+            mode: st.st_mode as u32,
```

Comment:
Should conditional compile with `#[cfg(target_os = "macos")]`.

Reply:
> https://doc.rust-lang.org/reference/expressions/operator-expr.html#type-cast-expressions

> Casting between two integers of the same size (e.g. i32 -> u32) is a no-op (Rust uses 2's complement for negative values of fixed integers)

From the reference, same size cast has no cost.

Reply:
No worry, that should be zero-cost abstraction.
The compiler is very very smart now, even smarter than my expectation:)